doc/rules/class_notation/static_private_method.rst

```diff
@@ -0,0 +1,44 @@
+==============================
+Rule ``static_private_method``
+==============================
+
+Converts private methods to ``static`` where possible.
```

Comment:
I'm not convinced that we should promote this rule, I rather avoid using static where possible.

if you want to have it in, go ahead with merge @Wirone